src/common/datatypes/Duration.h

```diff
@@ -125,10 +125,8 @@ struct Duration {
   }
 
   std::string toString() const {
-    std::stringstream ss;
-    ss << "P" << months << "M" << days() << "D"
-       << "T" << seconds << "S";
-    return ss.str();
+    return folly::sformat(
+        "P{}MT{}.{:0>6}000S", months, seconds + microseconds / 1000000, microseconds % 1000000);
```

Comment:
LGTM. just one concern: why is there 9 digits after the decimal point, I think 6 digits are enough.

Reply:
Yes, I just reference cypher.